### PR TITLE
Add more documentation to read coap block callback

### PR DESCRIPTION
**Pull request by szczys**
- stream: update doxygen for read block callback

### CODE REVIEW

File: src/coap_blockwise.c [hunk 1]
```diff
@@ -142,9 +142,9 @@ static enum golioth_status process_blockwise_uploads(struct golioth_client *clie
 {
     enum golioth_status status = GOLIOTH_ERR_FAIL;
     int ret = call_read_block_callback(ctx);
```
Comment (sam-golioth) [inline src/coap_blockwise.c:144]: This function does actually return a negative value, so the assert you hit must be caused by something else.

File: include/golioth/stream.h [hunk 2]
```diff
@@ -72,13 +72,18 @@ enum golioth_status golioth_stream_set_sync(struct golioth_client *client,
...
 /// @param block_idx The index of the block to fill
-/// @param block_buffer The buffer that this callabck should fill
+/// @param block_buffer The buffer that this callback should fill
 /// @param block_size (in/out) Contains the maximum size of block_buffer, and
-///        should be set to the size of data actually placed in block_buffer
+///        should be set to the size of data actually placed in block_buffer.
+///        Value must always be > 0.
 /// @param is_last (out) Set this to true if this is the last block to transfer
 /// @param arg A user provided argument
 typedef enum golioth_status (*stream_read_block_cb)(uint32_t block_idx,
```
Comment (sam-golioth) [inline include/golioth/stream.h:89]: Let's add information about the return value.

File: include/golioth/stream.h [hunk 3]
```diff
@@ -72,13 +72,18 @@ enum golioth_status golioth_stream_set_sync(struct golioth_client *client,
 
 /// Read block callback
 ///
-/// This callback will be called by the golioth client each time it needs to
+/// This callback will be called by the Golioth client each time it needs to
 /// fill a block.
 ///
+/// When the callback runs, the buffer should be filled with `block_size` number of bytes. When the
```
Comment (sam-golioth) [inline include/golioth/stream.h:78]: Let's wrap at 80 to match the rest of this file

File: include/golioth/stream.h [hunk 4]
```diff
@@ -72,13 +72,18 @@ enum golioth_status golioth_stream_set_sync(struct golioth_client *client,
...
+/// `last block` contains fewer bytes, set `block_size` to the actual number of bytes written. Set
+/// `is_last` to indicate all data has been written to buffer.
+///
 /// @param block_idx The index of the block to fill
-/// @param block_buffer The buffer that this callabck should fill
+/// @param block_buffer The buffer that this callback should fill
 /// @param block_size (in/out) Contains the maximum size of block_buffer, and
-///        should be set to the size of data actually placed in block_buffer
+///        should be set to the size of data actually placed in block_buffer.
+///        Value must always be > 0.
```
Comment (sam-golioth) [inline include/golioth/stream.h:86]:
```suggestion
///        Value must be > 0 when this function returns GOLIOTH_OK,
```

**Review by sam-golioth — CHANGES_REQUESTED** [review]

File: src/coap_blockwise.c (hunk 1 above)
Comment (szczys) [reply src/coap_blockwise.c:144]: Indeed, I tested this as working but if I return `GOLIOTH_OK` it asserts:

```
    if (block_idx == 10)
    {
        *block_size = 0;
        *is_last = true;
        return GOLIOTH_OK;
    }
```

**Review by sam-golioth — APPROVED** [review]

**Pull request merged**